File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
from collections import deque
n, m , v = map(int, sys.stdin.readline().split())

lst = [list(map(int,sys.stdin.readline().split())) for _ in range(m)]

graph = [[] for _ in range(n+1)]

for x,y in lst: 
  graph[x].append(y)
  graph[y].append(x)

for i in graph:
  i.sort()

def bfs(start):
  visited = [start]
  q = deque()
  q.append(start)
  while(q):
    cur_v = q.popleft()
    for v in graph[cur_v]:
      if(v not in visited):
        visited.append(v)
        q.append(v)
  return visited

visited = []
def dfs(cur_v):
  visited.append(cur_v)
  for v in graph[cur_v]:
    if v not in visited:
      dfs(v)
  return visited


  
print(*dfs(v))
print(*bfs(v))

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.js
const fs = require("fs");
// TODO: 제출 시 경로 변환 필수 ("/dev/stdin")
//2.14
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");
const [n, m, v] = input
  .shift()
  .split(" ")
  .map(e => Number(e));

graph = Array.from(Array(n + 1), () => []);
for (let i = 0; i < m; i++) {
  const [a, b] = input
    .shift()
    .split(" ")
    .map(e => Number(e));
  graph[a].push(b);
  graph[b].push(a);
}
graph.map(e => e.sort((a, b) => a - b));

let dfs_result = [];
const dfs = v => {
  dfs_result.push(v);
  for (let i of graph[v]) {
    if (!dfs_result.includes(i)) {
      dfs(i);
    }
  }
};

let bfs_result = [];
const bfs = v => {
  q = [v];
  bfs_result.push(v);
  while (q.length) {
    let cur = q.shift();
    for (let i of graph[cur]) {
      if (!bfs_result.includes(i)) {
        q.push(i);
        bfs_result.push(i);
      }
    }
  }
};

dfs(v);
bfs(v);
console.log(...dfs_result);
console.log(...bfs_result);
